src/connectors.rs

```diff
@@ -632,6 +632,7 @@ async fn connector_task(
                                 .addr
                                 .send(SourceMsg::Link {
                                     port,
+                                    tx: result_tx.clone(),
```

Comment:
you need to be careful here. Below in line 657 the result of the `send` operation of the `SourceMsg::Link` is sent to the `result_tx`. The result of the `Link` operation is not waited for, if done like this.
I'd suggest to only send a result to `result_tx` if we get into an errorcondition here. And in the case where the `Link` was correctly sent, do nothing, the one who created the `LinkOutput` msg will wait on its receiving side of the `result_tx` for the one message saying it is all done.